tests/python/openassetio/test_context.py

```diff
@@ -80,31 +80,30 @@ class TestState(managerAPI.ManagerStateBase):
 
 class Test_Context_access:
     def test_when_set_to_unknown_value_then_raises_ValueError(self, a_context):
-        with pytest.raises(ValueError) as err:
-            a_context.access = "unrecognized"
 
-        assert (str(err.value) ==
-                "'unrecognized' is not a valid Access Pattern"
-                " (read, readMultiple, write, writeMultiple, unknown)")
+        expected_msg = (r"incompatible function arguments.*\n"
+                        r".*arg0: openassetio._openassetio.Context.Access")
+
+        with pytest.raises(TypeError, match=expected_msg):
+            a_context.access = "unrecognized"
 
 
 class Test_Context_retention:
     def test_when_set_to_unknown_value_then_raises_ValueError(self, a_context):
-        with pytest.raises(ValueError) as err:
-            a_context.retention = 123
+        expected_msg = (r"incompatible function arguments.*\n"
+                        r".*arg0: openassetio._openassetio.Context.Retention")
 
-        assert (str(err.value) ==
-                "123 (123) is not a valid Retention (ignored, transient, session, permanent)")
+        with pytest.raises(TypeError, match=expected_msg):
+            a_context.retention = 123
 
 
 class Test_Context_locale:
     def test_when_set_to_unknown_value_then_raises_ValueError(self, a_context):
-        with pytest.raises(ValueError) as err:
```

Comment:
[fix] Can we add a test where we check that set and get of a valid `TraitsData` instance `is` the same, (to check we do accept that type, and that we're not copying by accident etc... re all the holder discussions)?

Reply:
Added test in aa39e3c1